master/internal/core.go

```diff
@@ -785,8 +785,11 @@ func (m *Master) Run(ctx context.Context) error {
 		m.echo.Use(auditLogMiddleware())
 	}
 
-	otelConfig()
-	m.echo.Use(otelecho.Middleware("determined-master"))
+	if m.config.Telemetry.OtelEnabled {
+		endpoint := m.config.Telemetry.OtelExportedOtlpEndpoint
+		otelConfig(endpoint)
```

Comment:
I also feel `otelConfig` should be `configureOtel` or something instead.
```suggestion
		otelConfig(m.config.Telemetry.OtelExportedOtlpEndpoint)
```